api/src/main/java/org/geysermc/geyser/api/block/custom/component/PlacementConditions.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2019-2022 GeyserMC. http://geysermc.org
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+*/
+
+package org.geysermc.geyser.api.block.custom.component;
+
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/*
+ * This class is used to store conditions for a placement filter for a custom block.
+ */
+public record PlacementConditions(@NonNull Set<Face> allowedFaces, @NonNull LinkedHashMap<String, BlockFilterType> blockFilters) {
+    public enum Face {
+        DOWN(1),
+        UP(2),
+        NORTH(4),
+        SOUTH(8),
+        WEST(16),
+        EAST(32);
+
+        Face(int value) {
+            value = (byte) value;
```

Comment:
Is this used anywhere?

Reply:
Well it's used directly above to declare the values for the enums without having to case each one to a byte

Reply:
Apologies I understand now and have removed this:
https://github.com/GeyserMC/Geyser/pull/3505/commits/f19e16f42a1f8493568c6e41a0195aa73ae66e49#diff-a78b23e9387790d8b766ab6542d07b8b4603bcec23349c247e53dd4ed24b6446